### PR TITLE
test: add `single-node` binary tests

### PR DESCRIPTION
**Pull request by kwannoel**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Also tests sqlbackend, since that's used by single-node mode.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by kwannoel** [comment]
Encounters `stack overflow` error.

**Comment by kwannoel** [comment]
Ready for review.

**Comment by kwannoel** [comment]
Hi @yezizp2012, looks like the tests are failing, could you PTAL? Thanks!

cc @huangjw806, only after this is merged in, then we should cut a new release for 1.7.0-standalone.

**Comment by kwannoel** [comment]
Here's my local errors when starting it:
```
cargo build --bin risingwave
./target/debug/risingwave
```

log
```
2024-02-28T23:34:08.04622+08:00  INFO rw-main risingwave_common::telemetry::report: start telemetry reporting
thread 'rw-main' panicked at src/meta/src/barrier/recovery.rs:348:14:
clean dirty streaming jobs: Query Error: error returned from database: (code: 1) no such column: sink.target_table

Backtrace:

```

**Comment by yezizp2012** [comment]
> Hi @yezizp2012, looks like the tests are failing, could you PTAL? Thanks!
> 
> cc @huangjw806, only after this is merged in, then we should cut a new release for 1.7.0-standalone.

Looks like the single node test passed in the ci workflow?


> Here's my local errors when starting it:
> 
> ```
> cargo build --bin risingwave
> ./target/debug/risingwave
> ```
> 
> log
> 
> ```
> 2024-02-28T23:34:08.04622+08:00  INFO rw-main risingwave_common::telemetry::report: start telemetry reporting
> thread 'rw-main' panicked at src/meta/src/barrier/recovery.rs:348:14:
> clean dirty streaming jobs: Query Error: error returned from database: (code: 1) no such column: sink.target_table
> 
> Backtrace:
> ```

It should be because you switched branches locally and ran different binaries but forgot to clean up the SQLite file. The old SQLite file was probably generated on a branch without the sink into table feature, while the `release-1.7.0-standalone` branch includes it.

**Comment by kwannoel** [comment]
> > Hi @yezizp2012, looks like the tests are failing, could you PTAL? Thanks!
> 
> Looks like the single node test passed in the ci workflow?
> 
> > Here's my local errors when starting it:
> > ```
> > cargo build --bin risingwave
> > ./target/debug/risingwave
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > log
> > ```
> > 2024-02-28T23:34:08.04622+08:00  INFO rw-main risingwave_common::telemetry::report: start telemetry reporting
> > thread 'rw-main' panicked at src/meta/src/barrier/recovery.rs:348:14:
> > clean dirty streaming jobs: Query Error: error returned from database: (code: 1) no such column: sink.target_table
> > 
> > Backtrace:
> > ```
> 
> It should be because you switched branches locally and ran different binaries but forgot to clean up the SQLite file. The old SQLite file was probably generated on a branch without the sink into table feature, while the `release-1.7.0-standalone` branch includes it.

Yes, now it works for me. Thanks!

File: ci/scripts/run-e2e-test.sh [hunk 1]
```diff
@@ -74,7 +88,9 @@ echo "--- e2e, $mode, batch"
 RUST_LOG="info,risingwave_stream=info,risingwave_batch=info,risingwave_storage=info" \
 cluster_start
 sqllogictest -p 4566 -d dev './e2e_test/ddl/**/*.slt' --junit "batch-ddl-${profile}"
-sqllogictest -p 4566 -d dev './e2e_test/background_ddl/basic.slt' --junit "batch-ddl-${profile}"
+if [[ "$mode" != 'single-node' ]]; then
+  sqllogictest -p 4566 -d dev './e2e_test/background_ddl/basic.slt' --junit "batch-ddl-${profile}"
```
Comment (yezizp2012) [inline ci/scripts/run-e2e-test.sh:92]: Will this e2e test fail in single node mode? 🤔

**Review by yezizp2012 — APPROVED** [review]
Rest LGTM

File: ci/scripts/run-e2e-test.sh (hunk 1 above)
Comment (kwannoel) [reply ci/scripts/run-e2e-test.sh:92]: Yes it fails, but I forgot why 🥵

**Pull request merged**